### ISSUE TITLE
Import type assist doesn't work near syntax errors

### ISSUE THREAD

**Issue by lnicola** [rust-analyzer/rust-analyzer#5145]
```rust
struct S;

impl Debug for S // need to add {} to be able to trigger the assist on Debug
```

Might be related to #5144.

**Comment by lnicola** [comment]
Looks like we fall through here: https://github.com/rust-analyzer/rust-analyzer/blob/aa69757a01c26cfad12498053c55cbc3d66a4bdb/crates/ra_assists/src/handlers/auto_import.rs#L97.

**Comment by lnicola** [comment]
Probably fixed in #5280.
